### PR TITLE
feat(components/packages): update axe-core if @skyux-sdk/testing installed

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2030/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2030/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2030/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2030/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2030/playground/)

File: libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
+
+import { resolve } from 'path';
+
+import { createTestLibrary } from '../../../testing/scaffold';
+
+describe('i18n-resources-module.schematic', () => {
```
Comment (johnhwhite) [inline libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.spec.ts]: This label seems incorrect.

File: libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * Sets axe-core to ~4.8.3. We can't put this in the ng-update.packageGroup
+ * because we only want to update a customer's axe-core version if they've
+ * installed @skyux-sdk/testing.
+ */
+export default function updateAxeCore(): Rule {
+  return (tree, context) => {
+    if (getPackageJsonDependency(tree, '@skyux-sdk/testing') !== null) {
+      const axeCore = getPackageJsonDependency(tree, 'axe-core');
+
+      if (axeCore !== null) {
```
Comment (johnhwhite) [inline libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.ts]: If they have testing but are missing the peer dependency, wouldn't that also be a good reason to install `axe-core`?

File: libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.spec.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.spec.ts]: Fixed.

File: libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.ts (hunk 2 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.ts]: Good call. Updated.

File: libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.ts [hunk 3]
```diff
@@ -13,16 +19,19 @@ import {
...
-        addPackageJsonDependency(tree, axeCore, 'package.json');
+      const packageJsonPath = '/package.json';
 
-        context.addTask(new NodePackageInstallTask());
-      }
+      removePackageJsonDependency(tree, AXE_CORE_PACKAGE, packageJsonPath);
+      addPackageJsonDependency(tree, axeCore, packageJsonPath);
     }
+
+    context.addTask(new NodePackageInstallTask());
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.ts:35]: Do we still need to add this task if we don't have anything to update?

Comment (Blackbaud-SteveBrush) [reply libs/components/packages/src/schematics/migrations/update-10/axe-core/axe-core.schematic.ts:35]: Yeah, in the cases where `@skyux-sdk/testing` is installed but `axe-core` is not (we install `axe-core` in this schematic if it's not installed already).

**Review by Blackbaud-CoreyArcher — APPROVED** [review]

**Pull request merged**